File: src/main/java/com/ccut/yiyi/service/StudentService.java
package com.ccut.yiyi.service;

import com.ccut.yiyi.model.Student;
import com.ccut.yiyi.model.group.StudentGroup;
import org.springframework.data.domain.Page;

import java.util.Map;

/**
 * @ClassName: StudentService
 * @Description: TODO
 * @Author: Mark
 * @Date: 2019/12/18 17:02
 * @version: V1.0
 */
public interface StudentService {
    /**
     * 查询选中社团所有学生信息
     * 条件查询+分页
     *
     * @param whereMap 查询条件
     * @param page     页码
     * @param size     每页显示条数
     * @param assoId   社团id
     * @return 返回学生分页数据
     */
    Page<StudentGroup> findSearch(Map whereMap, int page, int size, Integer assoId);

    /**
     * 添加社团成员
     *
     * @param studentGroup
     */
    void add(StudentGroup studentGroup);

    /**
     * 根据学生id查询一个学生信息
     *
     * @param id
     * @return
     */
    Student findById(Integer id);

    /**
     * 删除学生信息
     *
     * @param id     学生主键id
     * @param assoId 社团id
     */
    void deleteById(Integer id, Integer assoId);

    /**
     * 修改学生信息
     *
     * @param oldStuCode 修改前的学号
     * @param student    学生信息
     */
    void update(String oldStuCode, Student student);

    /**
     * 修改是否为此社团管理人员
     *
     * @param stuCode 学号
     * @param assId   社团id
     * @param sign    标记 1为修改为管理人员 0为取消管理员
     */
    void updateManage(String stuCode, Integer assId, Integer sign);

    /**
     * 通过学号查询学生信息
     * @param stuCode   学号
     */
    Student findByStuCode(String stuCode);
}

File: src/main/resources/static/js/service/studentService.js
//服务层
app.service('studentService', function ($http) {

    //读取列表数据绑定到表单中
    this.findAll = function () {
        return $http.get("/student/findAll");
    }
    //分页
    this.findPage = function (page, rows) {
        return $http.get('../student/findPage.do?page=' + page + '&rows=' + rows);
    }
    //查询一个学生的信息
    this.findOne = function (id) {
        return $http.get('/student/findOne/' + id);
    };
    //增加社团成员
    this.add = function (entity) {
        return $http.post('/student/add', entity);
    };
    //修改学生信息
    this.update = function (oldStuCode, po) {
        return $http.post('/student/update/' + oldStuCode, po);
    };
    //删除学生信息
    this.dele = function (id, assoId) {
        return $http.delete('/student/delete/' + id + "/" + assoId);
    };
    //修改是否为管理人员
    this.setManage = function (stuCode, assoId,sign) {
        return $http.post('/student/updateManage/' + stuCode + "/" + assoId+"/"+sign);
    };
    //查询此社团所有学生信息
    this.search = function (page, rows, assoId, searchEntity) {
        return $http.post('/student/search/' + page + "/" + rows + "/" + assoId, searchEntity);
    }
});
